### PR TITLE
Hub spoke deployment updates

### PR DESCRIPTION
**Pull request by neilpeterson**
Current deployment ([link](https://github.com/Azure/azure-quickstart-templates/tree/master/101-hub-and-spoke-sandbox)) is missing components from the RA (Azure Firewall) and also uses a virtual machine jump box.

This update adds the following:

- Azure Firewall
- Bastion (removed jump box)
- Option to deploy virtual machines into the first spoke

### CODE REVIEW

File: Solutions/azure-hub-spoke/nestedtemplates/azuredeploy-firewall.json [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            },
+            "properties": {
+                "publicIPAllocationMethod": "Static"
+            }
+        },
+        {
+            "type": "Microsoft.Network/azureFirewalls",
+            "apiVersion": "2020-05-01",
+            "name": "[parameters('firewallName')]",
+            "location": "eastus",
```
Comment (ckittel) [inline Solutions/azure-hub-spoke/nestedtemplates/azuredeploy-firewall.json:49]: hardcoded

File: Solutions/azure-hub-spoke/nestedtemplates/azuredeploy-vnet-peering.json [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            "type": "string"
+        },
+        "location": {
+            "type": "string"
+        }
+    },
+    "variables": {
+        "peering-name-hub-to-spoke-one": "hub-to-spoke-one",
+        "peering-name-hub-to-spoke-two": "hub-to-spoke-two",
+        "peering-name-spoke-to-hub-one": "spoke-one-to-hub",
```
Comment (ckittel) [inline Solutions/azure-hub-spoke/nestedtemplates/azuredeploy-vnet-peering.json:22]: Idea: `spoke-to-<hub name>`

File: Solutions/azure-hub-spoke/nestedtemplates/azuredeploy-bastion.json [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                            "protocol": "Tcp",
+                            "sourcePortRange": "*",
+                            "sourceAddressPrefix": "*",
+                            "destinationPortRange": "443",
+                            "destinationAddressPrefix": "AzureCloud",
+                            "access": "Allow",
+                            "priority": 120,
+                            "direction": "Outbound"
+                        }
+                    }
```
Comment (ckittel) [inline Solutions/azure-hub-spoke/nestedtemplates/azuredeploy-bastion.json:117]: Can block all remaining outbound traffic at this point.

File: Solutions/azure-hub-spoke/nestedtemplates/azuredeploy-virtual-network.json [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        "location": {
+            "type": "string"
+        }
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Network/networkSecurityGroups",
+            "apiVersion": "2019-11-01",
+            "name": "[parameters('nsgName')]",
+            "location": "[parameters('location')]"
```
Comment (ckittel) [inline Solutions/azure-hub-spoke/nestedtemplates/azuredeploy-virtual-network.json:29]: Should only allow SSH/Bastion in.

File: Solutions/azure-hub-spoke/nestedtemplates/azuredeploy-bastion.json (hunk 3 above)
Comment (neilpeterson) [reply Solutions/azure-hub-spoke/nestedtemplates/azuredeploy-bastion.json:117]: Just want to verify that I'm looking at this correctly. To me, it looks like a default deny all policy (both in and out) is applied when creating an NSG.

https://docs.microsoft.com/en-us/azure/virtual-network/network-security-groups-overview#denyalloutbound

Do you feel it's necessary to include in the template?

Comment (ckittel) [reply Solutions/azure-hub-spoke/nestedtemplates/azuredeploy-bastion.json:117]: explicit denys are added to bypass the two built-in allow policies.  Unless you have a reason to leverage those polices, then you don't need to short circuit the deny path.

Comment (neilpeterson) [reply Solutions/azure-hub-spoke/nestedtemplates/azuredeploy-bastion.json:117]: Thanks for the conversation.

**Pull request closed without merging**